bundles/redhat8.6/bundle.sh.gotmpl

```diff
@@ -13,57 +13,99 @@ RHSM_ACTIVATION_KEY=${RHSM_ACTIVATION_KEY:-""}
+  fi
 
-if [[ ${FOUND} == false ]]; then
-  echo "You must use subscription manager to fetch packages for redhat"
-  exit 1
-fi
+  if [[ -n "${RHSM_ORG_ID}" && -n "${RHSM_ACTIVATION_KEY}" ]]; then
+    subscription-manager register --org="${RHSM_ORG_ID}" --activationkey="${RHSM_ACTIVATION_KEY}" --force
+    FOUND=true
+  fi
 
-subscription::unregister() {
-  subscription-manager unregister
-}
+  if [[ -n "${RHSM_USER}" && -n "${RHSM_PASS}" && ${FOUND} == false ]]; then
+    subscription-manager register --username="${RHSM_USER}" --password="${RHSM_PASS}" --force
+    FOUND=true
+  fi
 
-subscription::defer_unregister() {
+  if [[ ${FOUND} == false ]]; then
+    echo "You must use subscription manager to fetch packages for redhat"
+    exit 1
+  fi
+
+  subscription::unregister() {
+    subscription-manager unregister
+  }
+
+  subscription::defer_unregister() {
     trap subscription::unregister ABRT
     trap subscription::unregister EXIT
     trap subscription::unregister HUP
     trap subscription::unregister INT
     trap subscription::unregister TERM
     trap subscription::unregister USR1
     trap subscription::unregister USR2
-}
+  }
+
+  subscription-manager release --set=8.6
+  subscription-manager refresh
+  subscription::defer_unregister
+
+  ENABLED_REPOS="codeready-builder-for-rhel-8-x86_64-rpms,rhel-8-for-x86_64-appstream-rpms,rhel-8-for-x86_64-baseos-rpms"
+  EUS_REPOS=${EUS_REPOS:-""}
+  if [[ -n "${EUS_REPOS}" ]]; then
+    #disables the standard repositories which should not be enabled when using EUS
+    subscription-manager repos --disable=rhel-8-for-x86_64-baseos-rpms --disable=rhel-8-for-x86_64-appstream-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-baseos-eus-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-appstream-eus-rpms
+    subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-eus-rpms
+    ENABLED_REPOS="codeready-builder-for-rhel-8-x86_64-eus-rpms,rhel-8-for-x86_64-appstream-eus-rpms,rhel-8-for-x86_64-baseos-eus-rpms"
+  else
+    subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-appstream-rpms
+    subscription-manager repos --enable rhel-8-for-x86_64-baseos-rpms
+  fi
+else
+  echo "Bypassing subscription-manager and upstream RHEL repositories due to --skip-subscription-manager being set to true."
+fi
+
+KUBERNETES_REPOS=${KUBERNETES_REPOS:-""}
+if [[ -n "${KUBERNETES_REPOS}" ]]; then
+  ENABLED_REPOS="${ENABLED_REPOS},kubernetes"
+fi
+
+if [[ -f /etc/yum.repos.d/user-repos.repo ]]; then
+  USER_REPOS="$(awk -F '[][]' '/^\[.*\]/ {print $2}' /etc/yum.repos.d/user-repos.repo | paste -sd, -)"
+  if [[ -n "${ENABLED_REPOS}" ]] && [[ -n "${USER_REPOS}" ]]; then
+      ENABLED_REPOS="${ENABLED_REPOS},${USER_REPOS}"
+  elif [[ -z "${ENABLED_REPOS}" ]] && [[ -n "${USER_REPOS}" ]]; then
+      ENABLED_REPOS="${USER_REPOS}"
+  fi
+fi
```

Comment:
I can think of another way of handling this. not a huge fan of usig awk this way

Reply:
@faiq I changed this in 6809b18. Let me know what you think.